crowbar_framework/lib/crowbar/upgrade.rb

```diff
@@ -34,6 +34,7 @@ def upgrade
       case @version
       when "1.9"
         knife_files_1_9
+        reset_target_platform_for_nodes
```

Comment:
I think [my previous comment about mixing refactoring with behavioural changes](https://github.com/crowbar/crowbar-core/pull/564#discussion_r72160108) still applies here, doesn't it? I'm struggling to understand why this addition is in this commit rather than in the "Patch upgrade data for 3.0 to 4.0 upgrades" commit (and even that one seems to be a mix of two different things).

BTW the main reason for being pedantic about how the commits are split is not (just ;-) ) to annoy you, but that I'm pretty ignorant in this area, so without each commit being small and with a clear commit message, I struggle to give a good review.